### PR TITLE
Boilerplate update

### PR DESCRIPTION
**Pull request by seanchoi-workfront**
This is for updating business-website with https://github.com/adobe/helix-project-boilerplate/blob/main/scripts/scripts.js

Related Jira ticket:
[MWPW-106514](https://jira.corp.adobe.com/browse/MWPW-106514)

Test URL:
https://boilerplate-update--business-website--adobe.hlx3.page/drafts/seanchoi/kitchen-sink

### CODE REVIEW

File: scripts/scripts.js [hunk 1]
```diff
@@ -819,165 +452,310 @@ export function addFavIcon(href) {
...
-    to: 'the-latest',
-  }, {
-    from: 'insights',
-    to: 'perspectives',
-  }];
-  replacements.forEach((r) => {
-    newpath = newpath.replace(`/${r.from}/`, `/${r.to}/`);
+async function waitForLCP() {
+  // eslint-disable-next-line no-use-before-define
+  const lcpBlocks = LCP_BLOCKS;
```
Comment (fullcolorcoder) [inline scripts/scripts.js:460]: Defined now and then later again? Also it's spitting up that LCP_BLOCKS isn't defined.

File: scripts/scripts.js [hunk 2]
```diff
@@ -819,165 +452,310 @@ export function addFavIcon(href) {
...
-export async function fetchPlaceholders() {
-  if (!window.placeholders) {
-    const resp = await fetch(`${getRootPath()}/placeholders.json`);
-    const json = await resp.json();
-    window.placeholders = {};
-    json.data.forEach((placeholder) => {
-      window.placeholders[placeholder.Key] = placeholder.Text;
+const RUM_GENERATION = 'project-1'; // add your RUM generation information here
+const PRODUCTION_DOMAINS = [];
+
```
Comment (davidnuescheler) [inline scripts/scripts.js:516]: LCP_BLOCKS would probably have to be added here, with the lcp blocks from https://github.com/adobe/business-website/pull/199/files#diff-fbceb58f32b99b8a5309251f8c49ac8eda82fba58b905c600c92bce1ed2a1c8cL1102

Comment (seanchoi-workfront) [reply scripts/scripts.js:516]: Even if waitForLCP() isn't used, do you still want to set LCP_BLOCKS?

File: scripts/scripts.js [hunk 3]
```diff
@@ -819,165 +452,310 @@ export function addFavIcon(href) {
...
-  const replacements = [{
-    from: 'news',
-    to: 'the-latest',
-  }, {
-    from: 'insights',
-    to: 'perspectives',
-  }];
-  replacements.forEach((r) => {
-    newpath = newpath.replace(`/${r.from}/`, `/${r.to}/`);
+async function waitForLCP() {
```
Comment (davidnuescheler) [inline scripts/scripts.js:458]: that's odd. the `waitForLCP()` is definitely required in the `loadEager()`

Comment (seanchoi-workfront) [reply scripts/scripts.js:458]: > that's odd. the `waitForLCP()` is definitely required in the `loadEager()`

I think their loadEager() is doing it similar thing itself.
image

Comment (seanchoi-workfront) [reply scripts/scripts.js:458]: I've updated to use waitForLCP() in loadEager() instead.

File: scripts/scripts.js (hunk 2 above)
Comment (davidnuescheler) [reply scripts/scripts.js:517]: ...waitForLCP has to be used. it blocks the display and other network activity before the LCP happened.

File: scripts/scripts.js (hunk 3 above)
Comment (davidnuescheler) [reply scripts/scripts.js:458]: i think you should remove that code and use `waitForLCP` instead, to be able to update things more easily going forward. there is no need to do this the old way, the `waitForLCP` code encapsulates this exact code in way that makes it easy to update.

**Comment by seanchoi-workfront** [comment]
image

@auniverseaway Could you take a look if this is also ok for you?

**Comment by auniverseaway** [comment]
My visual diff of the homepage is below.

The first thing that pops out to me is that the main title should be hidden. I just confirmed this with Porter.

business adobe com_blog_
boilerplate-update--business-website--adobe hlx3 page_blog_

**Comment by seanchoi-workfront** [comment]
@trieloff We have conflicts and I see your [PR #200](https://github.com/adobe/business-website/pull/200) that updating `sampleRUM()`. I was told to work on this from @davidnuescheler and not to touch anything above line 514:
image
Since `sampleRUM()` is in the "No touch" area, I think we should reconsider your part.
Though, `sampleRUM.sourceselector` can be still updated at line 553.

Please let me know what you think.

**Comment by trieloff** [comment]
@seanchoi-workfront  I've resolved the merge conflict, and made sure the RUM collection works as before.

**Review by trieloff — APPROVED** [review]
RUM stuff is ok

**Review by aaronmauchley-adobe — APPROVED** [review]
QA Approved.

**Comment by seanchoi-workfront** [comment]
Aaron has done with his QA:
"I tested various pages under blog and everything functioned and looked fine.  I did this under Chrome, Firefox, and Safari.  Marked the ticket as done."

I will merge this into main.

**Pull request merged**